### DISCUSSION TITLE
[FR] Add support for UI Elements to Matrix blocks

### DISCUSSION THREAD

**Discussion by daltonrooney** [craftcms/cms#8878] [Ideas]
### Description

I love the new UI elements feature, it would be nice to be able to add them within Matrix blocks as well as standard field layouts.

**Comment by piotrpog** [comment]
That would be pretty useful. Recently I created matrix field where each block had a date field, and when that date passed, block contents would not be shown on the frontend. If I could use UI elements within a block, i would be able to mark this block as "expired" in the control panel using a bit of twig code.

**Comment by outline4** [comment]
+1

**Comment by samstevens** [comment]
+1

**Comment by oles** [comment]
Been missing this for my last two CraftCMS projects :heart:

**Comment by Kashkin** [comment]
Would love to see this feature added. Any chance this will get added to an upcoming release?

**Comment by carlcs** [comment]
I have released a plugin with a couple of field types, which we’ve been using to add tips and warnings to Matrix blocks. It helps us bridge the waiting time until the feature gets implemented in Craft itself.

https://github.com/carlcs/craft-uielementfields
